Document/0x06e-Testing-Cryptography.md

```diff
@@ -2,12 +2,46 @@
+- [Heimdall](https://github.com/henrinormak/Heimdall "Heimdall"),
+- [SwiftyRSA](https://github.com/TakeScoop/SwiftyRSA "SwiftyRSA"),
+- [SwiftSSL](https://github.com/SwiftP2P/SwiftSSL "SwiftSSL"),
+- [RNCryptor](https://github.com/RNCryptor/RNCryptor "RNCryptor").
+
+##### Third party libraries
+There are various third party libraries available, such as:
+- CJOSE: With the rise of JWE, and the lack of public support for AES GCM, other libraries have found their way, such as [CJOSE](https://github.com/cisco/cjose "cjose"). CJOSE still requires a higher level wrapping as they only provide a C/C++ implementation.
+- CryptoSwift: A library in swift, which can be found at [Github](https://github.com/krzyzanowskim/CryptoSwift "CryptoSwift"). The library supports various hash-functions, MAC-functions, CRC-functions, symmetric ciphers, and password-based key derivation functions. It is not a wrapper, but a fully self-implemented version of each of the ciphers. It is important to verify the effective implementation of a function.
+- OpenSSL: [OpenSSL](https://www.openssl.org/ "OpenSSL") is the toolkit library used for TLS, written in . Most of its cryptographic functions can be used to do the various cryptographic actions necessary, such as creating (H)MACs, signatures, symmetric- & assymmetric ciphers, hashing, etc.. There are various wrappers, such as [OpenSSL](https://github.com/ZewoGraveyard/OpenSSL "OpenSSL") and [MIHCrypto](https://github.com/hohl/MIHCrypto "MIHCrypto").
+- LibSodium: Sodium is a modern, easy-to-use software library for encryption, decryption, signatures, password hashing and more. It is a portable, cross-compilable, installable, packageable fork of NaCl, with a compatible API, and an extended API to improve usability even further. See [LibSodiums documentation](https://download.libsodium.org/doc/installation "LibSodium docs") for more details. There are some wrapper libraries, such as [Swift-sodium](https://github.com/jedisct1/swift-sodium "Swift-sodium"), [NAChloride](https://github.com/gabriel/NAChloride "NAChloride"), and [libsodium-ios](https://github.com/mochtu/libsodium-ios "libsodium ios").
+- Tink: A new cryptography library by Google. Google explains its reasoning behind the library [in its security blog](https://security.googleblog.com/2018/08/introducing-tink-cryptographic-software.html "Introducing Tink"). The sources can be found at [Tinks Github repository](https://github.com/google/tink "Tink at Github").
+- Themis: a Crypto library for storage and messaging for Swift, Obj-C, Android/Java, С++, JS, Python, Ruby, PHP, Go. [Themis](https://github.com/cossacklabs/themis "Themis") uses LibreSSL/OpenSSL engine libcrypto as a dependency. It supports Objective-C and Swift for key generation, secure messaging (e.g. payload encryption and signing), secure storage and setting up a secure session. https://github.com/cossacklabs/themis/wiki/Objective-C-Howto!
```

Comment:
https://github.com/cossacklabs/themis/wiki/Objective-C-Howto! -> convert to clickable link and remove the exclamation point